File: 백준/Silver/1475. 방 번호/방 번호.kt
fun main() {
    val n = readln()
    val arr = Array(10) { 0 }

    var mx = 0
    for (c in n) {
        val num = c.digitToInt()

        arr[num]++
        mx = if (num == 6 || num == 9) {
            if ((arr[6] + arr[9]) % 2 == 1) {
                if ((arr[6] + arr[9]) / 2 + 1 <= mx) mx else (arr[6] + arr[9]) / 2 + 1
            } else {
                if ((arr[6] + arr[9]) / 2 <= mx) mx else (arr[6] + arr[9]) / 2
            }
        } else {
            if (arr[num] <= mx) mx else arr[num]
        }
    }

    println(mx)
}

File: 백준/Silver/1475. 방 번호/방 번호.cc
#include<iostream>
#include<algorithm>

using namespace std;

int arr[10];

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0); cout.tie(0);

    string n;
    cin >> n;

    int mx = 0;
    for (auto c : n) {
        int num = c - '0';

        arr[num]++;
        if (num == 6 || num == 9) {
            if ((arr[6] + arr[9]) % 2 == 1) {
                mx = mx > (arr[6] + arr[9]) / 2 + 1 ? mx : (arr[6] + arr[9]) / 2 + 1;
            } else {
                mx = mx > (arr[6] + arr[9]) / 2 ? mx : (arr[6] + arr[9]) / 2;
            }
        } else {
            mx = mx > arr[num] ? mx : arr[num];
        }
    }

    cout << mx;
}